consensus/solo/solo.go

```diff
@@ -202,18 +198,20 @@ func (self *SoloService) broadCrossShardHashMsgs(blkNum uint32, shardMsgs []xsha
 		shardMsgMap[msg.GetTargetShardID()] = append(shardMsgMap[msg.GetTargetShardID()], msg)
 	}
 	var hashes []common.Uint256
-	crossShardMsgs := &CrossShardMsgs{}
+	crossShardMsgs := &msg.CrossShardMsgs{}
 	for shardID, shardMsg := range shardMsgMap {
 		msgHash := xshard_types.GetShardCommonMsgsHash(shardMsg)
 		sig, err := signature.Sign(self.Account, msgHash[:])
 		if err != nil {
 			log.Errorf("sign cross shard msg failed, msg hash:%s, error: %s", msgHash.ToHexString(), err)
 			return
 		}
+		sigData := make(map[uint64][]byte)
+		sigData[self.shardID.ToUint64()] = sig
```

Comment:
use `0` instead since the key is node index , not shard id.

Reply:
SigData map[uint64][]byte ,this map key is node index

Reply:
had  modify SigData type map[uint32][]byte